### DISCUSSION TITLE
Discriminators for Revolt

### DISCUSSION THREAD

**Discussion by insertish** [stoatchat/discussions#61] [Feature Suggestions]
Do you think Revolt should have discriminators?

Discuss and argue below!

This has been turned into an RFC: https://github.com/revoltchat/rfcs/pull/3.

**Comment by sussycatgirl** [comment]
> (on the official instance)

Does this mean that this would be optional on selfhosted instances?

**Reply by infi** [reply]
This would mean a lot of extra work for developers of everything that connects to Revolt. All instances should use a unified naming scheme for users.

**Comment by ryanalexander** [comment]
Adding discriminators will make it difficult to introduce new users to the platform.

Users who have not used Discord are not used to discriminators, and for a good reason.
Unique names add a sense of accomplishment for those who register them, which can cause them to show off to new users who have not joined the platform yet.

**Reply by Kerkpower** [reply]
when i started using discriminators it was really easy to get used to id

**Reply by Ashe** [reply]
Actually, since discord and a lot of services now uses discriminators, you could argue that it's *easier* to introduce people to the platform.

Either they've used discord (which lets be honest, everyone interested in Revolt has), or, they just want to make a username. You just hide the discriminator until it comes to adding a friend, it's really simple.

If you reject new users because their preferred username isn't unique, they might take their time joining the server because they are trying to find a username they like. They might eventually forget and just go back to discord.

**Reply by Cyastis** [reply]
I would say it's neither easier nor harder to learn.. in fact, I would argue on ubiquitous applications, display names (where the UID/tag or whatnot is hidden) are more common. I don't find that discriminators are hard to remember or hard to teach to new users, but if they are, there are other solutions to easily add friends:

-friend invite link (like steam chat)
-proximity scan for mobile apps (like discord)
-search between users with the same name (like facebook, steam client)

Also, the same accomplishment feeling can also be achieved by having the #0001 (or else why would so many discord users go after this discriminator?)

**Comment by ffamilyfriendly** [comment]
Way easier to remember your username then having to remember your username and some numbers

**Reply by Ashe** [reply]
You use your email to log in anyway right? Same as Discord.

**Comment by aosync** [comment]
Discriminators make sense, but as @ryanalexander said, not having them gives a sense of accomplishment in owning the account. It gives them a fun uniqueness to the account and some known usernames could be seen as "legendary", even (_the_ insert, not an insert#xyzw)
I like not having them.

**Comment by Bossfuratu** [comment]
I know there's a whole argument out there about "claiming the name being an accomplishment" but it's more of an annoyance? It makes sense for most social media websites - which normally don't use discriminators, but Revolt (and Discord, which most of us should be familiar with) is a communication tool. Making an original username across everything you have can be hard enough and having a username that lines up with everything else seems important. Maybe there's other examples but I even recall Xbox ~~Live~~ Network moving to a system using discriminators.

**Comment by CaraAra** [comment]
Unique names carry an amount of both intimacy and simplicity.
When you interact with someone, it is not just user number 105641 (unless they explicitly chose to be 🙂😉), it is the person and the only one of themselves, just as they really are.

**Reply by Ashe** [reply]
Yeah, except you have to name yourself YourUsername105641, as opposed to that number being hidden by default. The only time I've ever cared about the discriminator in discord is adding someone who I'm not connected with yet - everyone else you just click their profile and add. When mentioning them, you just mention their username.

Discriminators have a small footprint with a big impact - without them, you have to go out of your way to have a username you like, which will typically manifest in longer names to type.

**Comment by Supersebi3** [comment]
i think having names be unique is really annoying for many people who want to use a simple, but consistent username across multiple platforms. discriminators are a wonderful way to solve this issue, and i really dont see any downside with them. its not that hard to remember 4 digits, and im sure new users can get used to it quickly - after all, it hasnt exactly hindered the adoption of discord. the whole "accomplishment" thing is entirely stupid - its your accont name, all you did was type something that nobody else happened to type before. it really doesnt mean anything and just makes adoption harder for later users because they will struggle to find a name

personally, i registered my revolt account as my discord username + my discord discriminator, because my usual username was already claimed

**Reply by Saragon4005** [reply]
Quite often 2 or 3 people may use the same username, who will likely never meet. I mean it's like having the same name as someone else, it's perfectly plausible, and it is also very unlikely that you will ever meet them.

**Comment by gdude2002** [comment]
Alright, I was asked to come here and leave a comment - [Insert right now](https://i.kym-cdn.com/photos/images/original/001/318/555/32b.jpg)

# Why?

As someone who's been moderating communities for around 16 years now, here's why I think discriminators are both useful and essential:

* The concept of the unique username is dead. It's been dead for a long time, and no amount of wishful thinking will bring it back. As the usability standards of (and, thus, audience for) any platform grows, more users will be attracted to it. We've seen this many times, especially in influencer circles. 
* Some names are trademarked or otherwise owned by companies - as we've seen, for example in the [npm "kik" package dumpster fire](https://arstechnica.com/information-technology/2016/03/rage-quit-coder-unpublished-17-lines-of-javascript-and-broke-the-internet/), those companies are likely to try to reclaim their names on any random platform they find themselves on. Limiting each name to a single use is just going to make this problem harder to deal with.
* As we've seen in the "rare name" jokes on the Revolt testers' server, short names and common words are harder names to grab. This is because there are both fewer to go around, and more people that are likely to be using these names. Limiting these names to a single use will encourage name squatting and people trying to sell accounts for common names, and will prevent people who normally have a consistent social media identity from using their names.

# Responses to other comments

## Discriminators are hard/I can't remember numbers

Discriminators are far from a new concept, and they are instinctively understood by many people - for those that don't immediately get it, there's no reason some simple documentation can't solve that problem. Additionally, in my experience, you seldom have to remember anyone's discriminator, including your own - platforms that use them (like Discord, Xbox and Blizzard's BNet) put them in very prominent places, and don't hide them, so there's really no reason to memorize them. This is even true for moderation, where you're likely to already be keeping logs regardless, and user IDs are more useful anyway.

In short, this isn't a valid argument - discriminators are a tried and tested way of multiple problems. They're easy to understand once you've understood why they exist, and I don't believe they make things any more difficult than usernames on their own do.

## There is prestige/accomplishment in a unique name

While this is entirely down to who you are as a person, consider that the chance that you end up with a name or not is essentially random on a platform with unique names. The sense of achievement you get in securing a name like `Apple` or `Dream` is essentially narcissistic - not that this is a bad thing, but to everyone else, you just kind of look self-centred when you brag about it.

On the other hand, if your "accomplishment" is in swiping brand names and common words early with the intention to sell them - well, there's a special place in hell for you, my friend.

Ultimately, whatever sense of accomplishment that a single user may end up feeling after grabbing an in-demand name is far less important than the frustration that people feel when they realize they can't use the name they've had since the 90s on your platform. The user that claimed it may even be impersonating them. It makes sense that some web-based social media networks like Twitter and GitHub don't use discriminators because they create profiles that you're expected to link to all over the internet - but a chat network like Revolt doesn't need this.

# Summary

A lot of what goes into a decision is personal preference from the founding developers - I get that. That said, I don't feel like that trumps what is a set of objective advantages for a discriminator system being in place. As long as users are unable to decide on their discriminator and Revolt makes it difficult to farm them, I think they'd work well.

And there's no reason Revolt can't get creative, if they'd like to - for example, hex-based discriminators, discriminator words, or even emoji - but I'll leave that up to them.

**Reply by Cyastis** [reply]
You have a point that unique usernames are really no longer feasible in the modern world. One has to look no further than Twitter, where any interesting handle name has been long taken, and hackers take over 3 letter handles and sell them for thousands, to see why that is. 

Why it may be novel in the beginning, after several tens of thousands of users join, it will no longer be feasible to acquire a good username and an even harder resistance (against discriminators, display names) will be felt from users that already have their "exclusive" usernames.

**Comment by ThatTonybo** [comment]
If we don't go for discriminators or any other choice, bots should be unique on their own level.

So, you could have a user and bot called Dave, but not two bots called Dave, or two users called Dave.

**Comment by doggogqgqgqgqgqgqgqgqgqg** [comment]
Nope Nope nope let people battle it out

**Comment by brecert** [comment]
Discriminators aren't strictly required to avoid completely unique usernames.
And unique usernames aren't the only alternative to discriminators.

When discussing ideas like this I think it's important to understand what benefits they should provide as a whole, not only when compared to another solution.

I haven't seen any discussion on why unique usernames, or discriminators are needed in the first place.
They have strengths, but they aren't exactly "needed".
Telegram for example doesn't require you to have a username at all if you don't want to set one.

In a good system, an account should be backed by a unique id that's separated from everything else anyways.

When I think about unique identifiers, there's two main reasons to use them
- easier to find the account and to share with others
- a familiar design, there's nothing new to learn for unique identifiers or discriminators

When I think unique identifiers have some disadvantages as well though too
- often too easy to abuse
- directly associated with the profile, making it harder to change without unintentionally affecting a lot
- too limited in scope (more of an opinion)

Similar to how servers have invite links, having profile codes could be alternative to having the identifier directly associated with the sharing part.

Also similar, they could have permission levels for the account and how they interact with it.
Ex. Having an account setting that allows dms from strangers by default, but disallows image sending unless the person comes from a specific profile code that allows them to send images.

Profile codes could easily be removed, added, or customized in appearance without altering the actual profile of the user.

In usage, I'm guessing profile codes would be similar to how discord discriminators work now for sharing a profile out.

You'd create one for you profile and it'd use your currently set display name for the identifier, and a unique code after, letting users adjust the length. Generating something like `revolt:@bree+7451` or however it was decided to be styled.

The benefits with this to me are that profile codes can be easily shared out and removed per-community to avoid harassment and to personalize embeds / previews per-community.

It avoids some of the issues with identifiers altogether such as having limited amounts by having the actual user account separated from the limited quantity problems, and user profiles can remain hidden and private and unshared for as long as they'd like to for those who wish to do that.

And it solves the disadvantages I see in unique identifiers as they are now, without losing many of their advantages.

**Reply by gdude2002** [reply]
Discriminators are, as I see it, a user-facing alternative to email addresses.

Many social media networks have accounts keyed by email address - Steam, for example. Steam accounts, however, have a separate unique ID (several of them for some reason). These IDs are never exposed, aside from in profile URLs (assuming the user doesn't have a custom URL). Because this is a bad way to represent yourself, Steam had to implement a display name search and a friend code system - there isn't really a great way to quickly find someone unless you know their name or profile link. It's far from ideal, I think you'd agree.

Discriminators solve this problem by allowing you to use the name you want, and combining it with a short number or sequence to make it unique - this has the advantage of allowing you to have a fairly easy to remember handle that you can give to people without worrying that it's cumbersome, whilst still allowing multiple people to have the same name.

---

I think your idea regarding profile codes _is_ interesting, though I feel like it'd be better solved by adding extra options when responding to a friend request - since currently you need to be friends with someone to DM them on Revolt. I don't see why you couldn't have "Accept", "Accept and allow images" and "Deny" there, for example.

I don't see long friend/profile codes as a solution to the problems that Discriminators solve, though - Nintendo in particular loves to throw friend codes around, and much of the community really doesn't like it. I'd argue that discriminators solve these problems in the most user-friendly way, at least that I've ever seen implemented.

**Reply by brecert** [reply]
I definitely agree that discriminators are the most user friendly / easiest way from what I've seen.

I'm not sure where you're getting long codes from, I mentioned how they'd likely look and function very similarly to how discriminators work already, just being detached from profile itself, and Nintendo's friend codes are a pain because they aren't able to be shortened, changed, and easily shared with others.

Profile codes like the way I described them in my idea avoids this problem, but you're right that it does come with a cost that it's not the most intuitive / easiest way of sharing a profile because you'd have to create one in the first place with the settings and whatnot.

Maybe profiles could have a few default codes (with common permissions) made already on creation so that it's easier to understand / use them.

Also, if you can think of any solutions outside of unique identifiers I think it'd be good to explore them, I think that unique identifiers are a flawed system that can be avoided but I only have so many ideas outside of them.

**Reply by gdude2002** [reply]
> I'm not sure where you're getting long codes from, I mentioned how they'd likely look and function very similarly to how discriminators work already

I guess I just kind of inferred it - though, I guess what you're saying is that someone could have several and give out the one they want people to have? I'm not entirely sure, haha

> Maybe profiles could have a few default codes (with common permissions) made already on creation so that it's easier to understand / use them.

That could be one approach I guess, yeah - although honestly, I'm not entirely sure in what situations you'd want to have separate settings for this. That said, it's certainly an interesting idea - basically invite links, but for DMs, maybe?

> Also, if you can think of any solutions outside of unique identifiers I think it'd be good to explore them, I think that unique identifiers are a flawed system that can be avoided but I only have so many ideas outside of them.

Ultimately, there has to be _some kind_ of unique identifier, from a technological standpoint. Internally, that's user IDs, of course - but I do think there's some value in being able to go "I'm gdude#2002 on Discord and gdude#1234 on Revolt". Having some kind of visible, usable unique identifier is very useful for things like identity verification, so I can't see it going away any time soon.

**Reply by brecert** [reply]
> I guess what you're saying is that someone could have several and give out the one they want people to have

Basically yeah. They'd be able to generate invite codes similar to discriminators, and share them out in the same way as you would with discord discriminators.

> basically invite links, but for DMs, maybe?

While DMs are a part of it, I'm considering the profile as a whole, similar to how telegram handles permissions like that.
- who can see your phone number
- who can see your history of avatars
- who can see your current avatar
- who can see when you were last online / currently online
- who can forward / quote your message
- who can add you to groups
- etc...

While these aren't all applicable to revolt, I think the idea of them is applicable.

Then when someone uses the profile discriminator they get added to that discriminators list and permission, similar to how roles might work with groups / channels.

> I'm not entirely sure in what situations you'd want to have separate settings for this

While generally at a certain size creating a group itself with links would be preferable, some people prefer privacy of their profiles (as I've seen with some on telegram). Unfortunately whitelisting / blacklisting each person manually isn't the best solution, with this role like system it'd be easy to change permissions / revoke them and group friend groups together.

A public twitter profile discriminator may want for only the current avatar to be seen by default while on a forum or in a discord guild a person may want for information to be seen.

This isn't mentioning the messaging restrictions / permissions which I think would make the most sense.

> Having some kind of visible, usable unique identifier is very useful for things like identity verification, so I can't see it going away any time soon.

I agree with that.
On the topic of identity verification, linking accounts would be neat if it's possible. Being able to use `discord:gdude#2002`, `twitter:brecert` or similar to find a user on revolt (if the user permits that of course) would be really nice / neat.

**Comment by LimeProgramming** [comment]
Yes to discriminators!

I agree with gdude above, so I'm not going to repeat what they said. 

I do honestly think it does just make things neater since people will wind up using discriminators in some way.
Think:
Dave1
Dave2
Dave3
Dave5
....
Dave4050 etc.

People generally only use their discriminator for friend requests, until then they are just eye sores. 

I suppose one could argue that there is some level of privacy to them. I go by CalamityLime on discord but that information alone isn't enough for grievers to spam me but it would tell people who I am if we happen in to be in similar circles. 
It's a bit of an edge case thing but worth pointing out. 

I think it's a no brainer. 
PS: Allowing people to edit discriminators just presents the "rare name" thing in a different format since everyone would fight for the lowest number or highest number. 
Rolling a dice and getting a really cool one by fluke is a great feeling.

**Reply by CaraAra** [reply]
Completely random discriminators are the worst option possible, but luckily, insert already confirmed they wouldn't be, so it's not even necessary to waste time arguing this.

**Reply by ryanalexander** [reply]
> Completely random discriminators are the worst option possible, but luckily, insert already confirmed they wouldn't be, so it's not even necessary to waste time arguing this.

When did he say this?

**Reply by CaraAra** [reply]
In the official public testers server a few days ago, you should be able to find it there.

**Reply by gdude2002** [reply]
I agree with @LimeProgramming, honestly. Randomized discriminators are the best approach because without them, you're just repeating the problems caused by forcibly unique usernames.

In fact, Discord didn't allow you to pick your discriminator for a super long time, and what happened? People started spamming accounts and renaming themselves to farm discriminators, like `#0001` or `#6666`. These accounts were often sold as well, and you often saw users with these discriminators holding it over other people's heads. Randomized discriminators and some kind of anti-farming logic definitely seems like the best option there.

In other cases, I understand how some people may use them as a sort of extension of their identity, though - I'm `gdude2002` here, and I have `gdude#2002` on Discord, for example. I think Discord was smart to only allow Nitro users to pick their discriminators, for that reason - it allows you to use the one you want, but the barrier to entry (payment) means that people are unlikely to use it just to get another sellable account, especially since you lose the custom discriminator if you stop paying.

**Reply by CaraAra** [reply]
You have ten thousand options to choose from, I promise you, noone will steal all the good ones connected to your name.
But the whole claim that discriminators allow people consistency with their other accounts elsewhere falls down if you force me to use #3507 while my tag elsewhere is #2226.
And just so you know, when people have to roll for something, they usually just roll as many times as they need to get a result they like, so you will not stop people making accounts with better tags, just make it convenient to spam create unused accounts.

**Reply by gdude2002** [reply]
Discord did manage to solve the "let's spam a bunch of accounts" problem, so I don't believe you can't do anything about that. It's always worth remembering that every approach is a tradeoff - there's no perfect solution here.

> But the whole claim that discriminators allow people consistency with their other accounts elsewhere falls down if you force me to use #3507 while my tag elsewhere is #2226.

Your discriminator on another chat network is far less important than being able to call yourself "gdude" - the consistency is about identity, and discriminators are not something people use to refer to you in chat. That's very much a false equivalence.

**Reply by CaraAra** [reply]
As long as it is a permanent, visible part of my profile, it is as important to me as every other part, it does not matter to me that you don't agree, there are no false equivalencies in personal value lists.

**Reply by gdude2002** [reply]
While that is true, my thought process is based on practicality rather than my personal preference - I'd enjoy having matching discrims as much as the next guy, but I don't think it's a practical concern for most people. 

I find that these types of discussions work best when one disregards one's personal preference and prioritizes what's most likely to work for the majority of people. It's also a great way to avoid bikeshedding!

**Reply by CaraAra** [reply]
I believe that what will work best for most people is the freedom to modify as much of their profile identity and ux as is possible to let them, including profile tags if those get introduced.

**Reply by gdude2002** [reply]
Alright, if you're sure - I'm happy to agree to disagree, haha

**Reply by pluiedev** [reply]
Yea for discriminators - I'm leocth31 on gmail *solely* because leocth is already taken. Thus there must be some other leocth somewhere who registered that account earlier than I did, which is good for them, but sucks for me. I also support random discriminators because that way no one will race to get special discriminators like 0001, 0069, 0420, etc.

**Reply by LimeProgramming** [reply]
Locking the editing of a discriminator behind a paywall was a solution that I remember annoying a lot of people when it came out but I thought it made practical sense. Helps avoid abuse of the rare names thing. 

Personally I hate Discords marketing, it feels so hollow and disingenuous which never helps for selling something practical. That's beside the point though. 

I'd really like to suggest something practical but I can't think of anything that's not too complex of a solution to a simple problem. Emojis would be different but a pain to enter, packs each with a theme would be mute if you can't edit your discriminator. Although with saying that I wouldn't mind being Lime@CalamityGanon or something silly like that. 
Numbers are just the most straightforward and easiest to type on every platform I've used.

**Reply by Yu-Vitaqua-fer-Chronos** [reply]
A two word combination system for discriminators seems like it'd be ideal to be honest, as it can be easy to be remember, I'd love to have my account be `Hikari#CreativePotato` lol, maybe having a sort of selection menu for a combination of words would be nice? It'd solve the argument of discriminators being hard to remember, that people bring up

- Hikari

**Reply by gdude2002** [reply]
I think a word system could work, but you'd need to make sure everyone with the same username had a unique combination - which is a lot harder than it is with numbers. You've also got to remember the spellings! Unless there was an autocomplete for them, I suppose?

**Reply by Yu-Vitaqua-fer-Chronos** [reply]
I'd assume there'd be a list of words if a word-combination system was used aha

Also could you explain what you mean by it's harder then if Revolt was using a number-based system instead? You'd need to still use code to check if the combination already exists for that username, whether it's words or numbers :p

And depending on the words I don't think spelling would be too hard to remember, especially if they're common words aha

- Izumi

**Reply by brecert** [reply]
I think a word system may be good because unlike numbers, names can be continually added, names aren't limited to 9999, and with each additional name choice it increases more than just another choice. 
Ex. `100 * 100 -> 100 * 101` adds another hundred combinations available by only adding a single name.

I quite like [Animal Crossing's selection of titles](https://nookipedia.com/wiki/Passport#Title) to use and how many of them are unlocked after certain milestones, maybe similarly *some* milestones on revolt could add additional words to use?
Ex. Donating, being a translator, being a contributor, being a theme contributor, etc..

Animal Crossing's current selection while limited already has 78391 combinations possible.
I don't think Animal Crossing's selection would fit for Revolt, but hopefully it gets my point / idea across.

However, a word system has it's drawbacks.

For one, I'm not sure how that'd work internationally. Not every language can identify or write latin letters while arabic numerals are pretty universal.

Another issue is that names will take up much more space `bree:LongtimeFan` may not fit in some bios. I know it shouldn't be too much of an issue but it certainly exists I think.

> make sure everyone with the same username has a unique combination , which is a lot harder than it is with numbers

And this is another one, but I think it could easily be solved by having a jump table / map from the strings to numbers, then looking up the numbers. It should be performant enough even if it's a little more complicated than looking up numbers alone

**Reply by brecert** [reply]
There could be an emoji system, but I find emojis to be noisy and hard to read.
They are a little more universal than english names / words though.

**Comment by qwertyquerty** [comment]
I vote no for discriminators. Let people own their username that no one else has.

**Comment by ghost** [comment]
This is a chat app, not MMO RPG. There is no point in having any accomplishments here. I like having my unique and possibly rare name but why should I take it from other people forever?

Discriminators is not some hard to understand concept. First time I had to deal with them I managed to get why is that a thing in 10 or so seconds. 

Even if someone isn't already familiar with discriminators yet, it doesn't matter. The most important thing is this person should be able to use their preferred username when registering in Revolt, no matter what.

**Comment by TaiAurori** [comment]
Something possibly of notation is how Xbox does it with the relatively recent addition of discriminators:
When someone tries to claim a gamertag and nobody else has that gamertag, Xbox will give the user *the canonical* gamertag. Then, when someone else tries claiming that gamertag, only then will the Xbox system assign them a discriminator.
A more general example: Tom 1 wants the username "Tom". Because nobody else has it, they are given the username that everyone can reach them at:`@Tom`. Now, Tom 2 tries to take the username "Tom", but because the canonical username is already given to Tom 1, Tom 2 is notified that their username will be given a discriminator, and Tom 2 gets `@Tom#2493`
I'm not saying this is directly *the best approach*, there are definitely some cons to this system, but this system is already used in practice, and could be a plausible option.

**Reply by gdude2002** [reply]
It's an interesting idea, but I feel like this isn't feasible for something like Revolt. This is mainly because a lot of people use social networks as points of contact - if you claim `TaiAurori` then it's not safe to say things like `I'm TaiAurori on Revolt` in permanent places - because as soon as someone else uses the name, you have to update everything (or people won't be able to contact you).

**Reply by LimeProgramming** [reply]
Could be fun. Even if that's just something for people who already have accounts with any new account getting discriminators. 
Something like an early bird badge

**Reply by Plecra** [reply]
Just to clarify here - the original `@Tom`'s username is never changed - you either have the canonical username, or came after and get a discriminator

**Comment by Ashe** [comment]
I really really really like discriminators - its 2021, I don't think usernames are important anymore. If you're going down the 'accomplishments' route, surely a badge system keeping track of when you joined, messages sent etc will be more 'impressive'? 

I don't know, but allowing only 1 person to achieve the 'accomplishment' of having their preferred username is a detrimental experience to other people

**Comment by dtupper** [comment]
Yes. Absolutely required.

Squatters are horrible for any platform and anything that can be done to discourage them is ideal.

My name is apparently already squatted on the official instance of Revolt, so... wonderful.

**Comment by super-continent** [comment]
discriminators are an absolute MUST imo. requiring unique usernames creates a lot of problems for people with common words as usernames and makes it annoying for users joining after the platform has existed for a while to find a username that isnt taken. I dont think theres a good reason to not use them. the "prestige" argument people seem to be using doesnt really make sense to me, names, like in real life, should not be a limited resource, anyone should be able to use any name without issue.

**Comment by peepo5** [comment]
I think a lack of discriminators is cool. Why? because then you know that people are unique and they are easily identifiable. Only issue is username spoofing with alternate characters that may need to be looked to. If I were you guys, i would limit usernames to [a-z], [A-Z], [0-9], [- _ + = ! @ # $ % ^ & * ( ) < > ? : ; " ' [ ] { } \ | ` (tilde) ]

Basically, all of the common type-able symbols.

**Reply by Saragon4005** [reply]
So twitter usernames? Why? There is a 99% chance you will never come across the person who made you put a 1 after your username. This isn't a global platform. Its way more scattered then even the least traditional social media. The only time the username *actually* matters is during freind requests and even that would be better with a link or QR code.

Usernames used to matter a lot because data was tied to the username. Nowdays it's the ID and the username is just a display name. Almost completely irrelevant.

**Reply by peepo5** [reply]
hmm :thinking:

**Reply by peepo5** [reply]
maybe discriminators from 0-99 then?

**Comment by Takanu** [comment]
There's too many people on the planet to think that unique usernames make a lot of sense - discriminators empower people more to name themselves as they'd like, the commodification of usernames shouldn't be seen as a plus when that kind of design encourages hacking and profile trading.

**Comment by DeMoN100** [comment]
Use discriminators it's a good system for identification.

### Why?
As a communication chat platform people always flock to what's new & try to take known brand names or common words. People name squat or attempt to sell usernames this is known people do this on discord, people sell custom.gg urls on guilded. This practice has been around for many years.

These kind of acts are narcissistic which isn't bad but can make you look self-centered when your the guy who's got 1000 usernames to sell. If your accomplishment is to take common words or brand names like apple, Microsoft, steam, master, element, universe, sensei. You really need to find something better to do.

The next thing people attempt to do is take popular youtubers names to cause defamation of character or pretend that youtuber is advertising their channel. Discriminators are a way to tell apart who's the real content creator from the person pretending to be him.  Guilded from example has shown how not having discriminator can be used in a bad way. I made 5 accounts with the same username on there to test it to see.

As revolt attracts a mix of people there are a lot of underage kids who like to troll when they are bored. Masquerading as someone else may ease them a bit but there are people who will try to take it a step further.

Discriminators can also make finding someone to add easier, likewise with searching for a message in a channel when you could just search the #xxxx  & bring up all the messages of that Id in a channel.

**Reply by nidmb** [reply]
Revolt doesn't allow for identical usernames though, no? There's no space for impersonation here.

**Reply by nulldg** [reply]
> Revolt doesn't allow for identical usernames though, no? There's no space for impersonation here.

name squatting gives rise to impersonation, which is mainly a problem for trademarks, artists, e-thots, usual targets of harassment, and so on. even without discrims, allowing identical usernames in itself would at least mean a username doesn't make somebody special.

**Comment by nidmb** [comment]
Yes and no. If you can implement it better than Discord did, go ahead.

However I still believe that display names combined with unique usernames are a better idea for a few reasons:
1. Display names allow everyone to display their own identity.
2. Unique human-generated usernames are more likely to be easily memorizable. It's easier to tell them apart too. This also reduces chances of impersonation.

**Reply by nulldg** [reply]
> Display names allow everyone to display their own identity.

i 100% fully support this.
hell, we don't even have to choose one or the other: display names *and* discrims would solve all the problems with both at the same time. with display names, discrims would only be relevant where absolutely necessary, eg adding friends from other sites, which is exactly where squatting (and thus impersonation) is a problem. for clarification, see [my previous comment](https://github.com/orgs/revoltchat/discussions/61#discussioncomment-5917285).


>This also reduces chances of impersonation.

only if you're lucky enough to get your name before impersonators do. if you're unlucky, it becomes an issue which is damn-near impossible for users to solve without begging/threatening revolt to help you (depending on whether you're a human or a corporate skinwalker).

discrims supposedly make impersonation less effective because it defeats the notion of name ownership, as there is no need to compete for them. this is additionally true if display names are added.

**Comment by rethyria** [comment]
I've read through a bunch of the comments above and just want to call out how to dumb the "sense of accomplishment" argument is.

Honestly, it reads like that infamous Reddit post from EA on loot boxes.

Sure, one person might get to feel smug about getting their preferred username, but for every person like that you might have 10s or 100s or 1000s of people disappointed to not get their preferred name.

**Comment by aurkaxi** [comment]
+1 for discriminator 

people already said the points i would say, so skipping.

**Comment by KotoWhiskas** [comment]
+ 1, and since discord is removing their own discriminators, this could help revolt's hype

**Comment by insertish** [comment]
Closing off this discussion since we have an open RFC for this now: https://github.com/revoltchat/rfcs/pull/3.
Please move further discussion over there.

**Comment by insertish** [comment]
RFC has been merged and implementation will land in Revolt soon.
